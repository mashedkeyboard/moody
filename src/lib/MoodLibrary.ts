import { Mood } from "$lib/models/Mood";

/** The default moods in the application. */
let moods = [new Mood(5, "😃"), new Mood(4, "🙂"), new Mood(3, "😐"), new Mood(2, "🙁"), new Mood(1, "😭")];

/** Gets a mood by its ID. */
function getMoodById(id: number) : Mood | null {
    let foundMood : Mood | null = null;

    moods.forEach(function (mood) {
        if (mood.id == id) foundMood = mood;
    });

    return foundMood;
}

export {moods, getMoodById};
