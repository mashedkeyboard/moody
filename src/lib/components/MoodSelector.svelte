<script lang="ts">
	import type { Mood } from "../models/Mood";
    import {moods} from "$lib/MoodLibrary";
    import { createEventDispatcher } from 'svelte';

	const dispatch = createEventDispatcher();

    function selectMood(mood: Mood) {
        dispatch('moodSelect', mood);
    }
</script>

<div class="flex flex-wrap justify-center align-center p-10 gap-3">
    {#each moods as mood}
    <button class="btn btn-circle btn-lg text-4xl" on:click={e => selectMood(mood)}>{mood.icon}</button>
    {/each}
</div>