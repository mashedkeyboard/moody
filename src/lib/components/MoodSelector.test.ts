import MoodSelector from '$lib/components/MoodSelector.svelte';
import type { Mood } from '$lib/models/Mood';
import { moods } from '$lib/MoodLibrary';
import { cleanup, fireEvent, render, screen } from '@testing-library/svelte'

describe('MoodSelector.svelte', () => {
  // TODO: @testing-library/svelte claims to add this automatically but it doesn't work without explicit afterEach
  afterEach(() => cleanup())

  it('mounts', () => {
    const { container } = render(MoodSelector, {})
    expect(container).toBeTruthy()
    expect(container.innerHTML).toContain('ðŸ˜­')
    expect(container.innerHTML).toMatchSnapshot()
  })

  moods.forEach(function(mood) {
    it(`dispatches an event when ${mood.icon} is selected`, async () => {
      const { component } = render(MoodSelector, {})
      const btn = screen.getByText(mood.icon);

      let moodSelected : Mood;
      const moodSelectedEvent = vi.fn((event) => moodSelected = event.detail);

      component.$on("moodSelect", moodSelectedEvent);
      await fireEvent.click(btn);

      expect(moodSelectedEvent).toHaveBeenCalled();
      expect(moodSelected).toBe(mood);
    })
  })
})
