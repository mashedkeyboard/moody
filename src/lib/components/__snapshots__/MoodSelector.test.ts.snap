// Vitest Snapshot v1

exports[`MoodSelector.svelte > mounts 1`] = `"<div><div class=\\"svelte-78a4lf\\"><button class=\\"round large svelte-114d4g7\\" type=\\"button\\">😃</button><button class=\\"round large svelte-114d4g7\\" type=\\"button\\">🙂</button><button class=\\"round large svelte-114d4g7\\" type=\\"button\\">😐</button><button class=\\"round large svelte-114d4g7\\" type=\\"button\\">🙁</button><button class=\\"round large svelte-114d4g7\\" type=\\"button\\">😭</button></div></div>"`;
